src/main/java/ghbot/GhBot.java

```diff
@@ -0,0 +1,56 @@
+package ghbot;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
+import ghbot.exception.GhBotException;
+import ghbot.executor.ByeExecutor;
+import ghbot.executor.Executor;
+import ghbot.parser.Parser;
+import ghbot.storage.Storage;
+import ghbot.task.TaskList;
+import ghbot.ui.Ui;
+
+/**
+ * GhBot Class.
+ * This is the main class for the chatbot.
+ */
+public class GhBot {
+    private Storage storage;
+    private TaskList taskLst;
+    private Ui ui;
+
+    /**
+     * GhBot Constructor.
+     * @param filename The name of the file.
+     */
+    public GhBot(String filename) {
+        this.storage = new Storage(filename);
```

Comment:
Perhaps could omit the 'this' keyword? I previously asked my TA about it and she mentioned this:
I got a question from one of your classmates regarding the this keyword. Use of this is discouraged, potentially to avoid unnecessary noise in the code. Try to avoid using this except when it is shadowed by a method or constructor parameter. This is an advanced rule that is optional to follow for CS2103T.